modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -2345,144 +2381,200 @@ private Expression<?>[] _getSelectExpressions(
+				}
 
-				predicate = relatedObjectDefinitionColumn.eq(
-					dynamicObjectDefinitionTable.getPrimaryKeyColumn());
+				selectExpressions.add(
+					DSLQueryFactoryUtil.scalarSubDSLQuery(
+						joinStep.where(predicate),
+						DynamicObjectDefinitionTable.getJavaClass(
+							objectField.getDBType()),
+						objectField.getName(),
+						DynamicObjectDefinitionTable.getSQLType(
+							objectField.getDBType())));
 			}
-			else if (Objects.equals(
-						objectRelationship.getType(),
-						ObjectRelationshipConstants.TYPE_MANY_TO_MANY)) {
-
-				DynamicObjectRelationshipMappingTable
-					dynamicObjectRelationshipMappingTable =
-						new DynamicObjectRelationshipMappingTable(
-							objectDefinition.getPKObjectFieldDBColumnName(),
-							relatedObjectDefinition.
-								getPKObjectFieldDBColumnName(),
-							objectRelationship.getDBTableName());
-
-				Column<DynamicObjectRelationshipMappingTable, Long>
-					primaryKeyColumn2 =
-						dynamicObjectRelationshipMappingTable.
-							getPrimaryKeyColumn2();
-
-				joinStep = joinStep.innerJoinON(
-					dynamicObjectRelationshipMappingTable,
-					primaryKeyColumn2.eq(
-						relatedDynamicObjectDefinitionTable.
-							getPrimaryKeyColumn()));
-
-				Column<DynamicObjectRelationshipMappingTable, Long>
-					primaryKeyColumn1 =
-						dynamicObjectRelationshipMappingTable.
-							getPrimaryKeyColumn1();
-
-				predicate = primaryKeyColumn1.eq(
-					dynamicObjectDefinitionTable.getPrimaryKeyColumn());
-			}
-
-			List<String> oDataFilterStrings = TransformUtil.transform(
-				(List<ObjectFilter>)objectFieldSettingsValues.get("filters"),
-				objectFilter -> {
-					if (StringUtil.equals(
-							objectFilter.getFilterType(),
-							ObjectFilterConstants.TYPE_CURRENT_USER)) {
-
-						objectFilter.setJSON(
-							JSONUtil.put(
-								"currentUserId",
-								PrincipalThreadLocal.getUserId()
-							).toString());
+			else if (objectField.compareBusinessType(
+						ObjectFieldConstants.BUSINESS_TYPE_FORMULA)) {
+
+				List<ObjectField> objectFields =
+					_objectFieldLocalService.getObjectFields(
+						objectDefinition.getObjectDefinitionId());
+
+				Map<String, Object> columnMap = new HashMap<>();
+
+				for (ObjectField objectField2 : objectFields) {
+					if (Objects.equals(
+							objectField2.getBusinessType(),
+							ObjectFieldConstants.BUSINESS_TYPE_ATTACHMENT) ||
+						Objects.equals(
+							objectField2.getBusinessType(),
+							ObjectFieldConstants.BUSINESS_TYPE_RICH_TEXT)) {
+
+						continue;
 					}
 
-					ObjectFilterParser objectFilterParser =
-						_objectFilterParserServiceRegistry.
-							getObjectFilterParser(objectFilter.getFilterType());
+					Column<?, Object> column =
+						(Column<?, Object>)_objectFieldLocalService.getColumn(
+							objectDefinition.getObjectDefinitionId(),
+							objectField2.getName());
 
-					return objectFilterParser.parse(objectFilter);
-				});
+					columnMap.put(objectField2.getName(), column);
```

Comment:
I see what Carol is saying. Great catch :clap: 
You populate the variable columnMap with all the object fields that belongs to the Object Definition.
If the Object Definition contains 2-3 formula field, the code will create/calculate the same columnMap over and over again. But, there is no need to populate again since the object fields would be the same to all the formula fields since we are facing the same Object Definition.

Reply:
Makes sense! thanks girls :+1: